eng/Dockerfile.examples

```diff
@@ -0,0 +1,55 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+ARG RELEASE=jammy
+
+FROM buildpack-deps:buster-scm AS buster
+
+RUN wget https://packages.microsoft.com/config/debian/10/packages-microsoft-prod.deb \
+    && dpkg -i packages-microsoft-prod.deb \
+    && rm packages-microsoft-prod.deb \
+    && apt-get update \
+    && apt-get install -y --no-install-recommends powershell python3-minimal python3-pip python3-setuptools sudo \
+    && rm -rf /var/lib/apt/lists/*
+
+FROM buildpack-deps:bullseye-scm AS bullseye
+
+RUN curl https://packages.microsoft.com/keys/microsoft.asc | apt-key add - \
+    && echo "deb [arch=amd64] https://packages.microsoft.com/repos/microsoft-debian-bullseye-prod bullseye main" > /etc/apt/sources.list.d/microsoft.list \
+    && apt-get update \
+    && apt-get install -y --no-install-recommends powershell python3-minimal python3-pip python3-setuptools sudo \
+    && rm -rf /var/lib/apt/lists/*
+
+FROM buildpack-deps:focal-scm AS focal
+
+RUN wget "https://packages.microsoft.com/config/ubuntu/20.04/packages-microsoft-prod.deb" \
+    && dpkg -i packages-microsoft-prod.deb \
+    && rm packages-microsoft-prod.deb \
+    && apt-get update \
+    && apt-get install -y --no-install-recommends powershell python3-minimal python3-pip python3-setuptools sudo \
+    && rm -rf /var/lib/apt/lists/*
+
+FROM buildpack-deps:jammy-scm AS jammy
+
+RUN wget "https://packages.microsoft.com/config/ubuntu/22.04/packages-microsoft-prod.deb" \
+    && dpkg -i packages-microsoft-prod.deb \
+    && rm packages-microsoft-prod.deb \
+    && apt-get update \
+    && apt-get install -y --no-install-recommends powershell python3-minimal python3-pip python3-setuptools sudo \
+    && rm -rf /var/lib/apt/lists/*
+
+FROM $RELEASE
+
+ARG USERNAME=pyqir
```

Comment:
@samarsha Have you tried overriding these vars (`USERNAME`, `USER_ID`, `USER_GID`) with `--build-arg` to make sure they are still applied?

Reply:
Yes, the build args still work.